flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkBlueGreenDeploymentController.java

```diff
@@ -0,0 +1,703 @@
+                        .filter(
+                                e ->
+                                        badEventPatterns.stream()
+                                                .anyMatch(
+                                                        p ->
+                                                                e.getReason()
+                                                                        .toUpperCase()
+                                                                        .contains(p)))
+                        .collect(Collectors.toList());
+
+        if (!abnormalEvents.isEmpty()) {
+            LOG.warn("Abnormal events detected: " + abnormalEvents);
+        }
+    }
+
+    private static Savepoint configureSavepoint(
+            FlinkResourceContext<FlinkDeployment> resourceContext) throws Exception {
+        // TODO: if the user specified an initialSavepointPath, use it and skip this?
+        Optional<Savepoint> lastCheckpoint =
+                resourceContext
+                        .getFlinkService()
+                        .getLastCheckpoint(
+                                JobID.fromHexString(
+                                        resourceContext
+                                                .getResource()
+                                                .getStatus()
+                                                .getJobStatus()
+                                                .getJobId()),
+                                resourceContext.getObserveConfig());
+
+        // Alternative action if no checkpoint is available?
+        if (lastCheckpoint.isEmpty()) {
+            throw new IllegalStateException(
+                    "Last Checkpoint for Job "
+                            + resourceContext.getResource().getMetadata().getName()
+                            + " not found!");
+        }
+        return lastCheckpoint.get();
+    }
+
+    private UpdateControl<FlinkBlueGreenDeployment> initiateDeployment(
+            FlinkBlueGreenDeployment bgDeployment,
+            FlinkBlueGreenDeploymentStatus deploymentStatus,
+            DeploymentType nextDeploymentType,
+            FlinkBlueGreenDeploymentState nextState,
+            Savepoint lastCheckpoint,
+            Context<FlinkBlueGreenDeployment> josdkContext,
+            boolean isFirstDeployment)
+            throws JsonProcessingException {
+
+        deploy(bgDeployment, nextDeploymentType, lastCheckpoint, josdkContext, isFirstDeployment);
+
+        setAbortTimestamp(bgDeployment, deploymentStatus);
+
+        return patchStatusUpdateControl(
+                        bgDeployment, deploymentStatus, nextState, JobStatus.RECONCILING)
+                .rescheduleAfter(getReconciliationReschedInterval(bgDeployment));
+    }
+
+    private boolean isDeploymentReady(
+            FlinkDeployment deployment,
+            Context<FlinkBlueGreenDeployment> josdkContext,
+            FlinkBlueGreenDeploymentStatus deploymentStatus) {
```

Comment:
We should not check for pods here, we can eliminate all this logic I believe. This is too low level on this controller and if we feel that some info is missing from the FlinkDeployment status then it should be added there